datasets/flwr_datasets/partitioner/natural_id_partitioner.py

```diff
@@ -87,3 +151,17 @@ def partition_id_to_natural_id(self, value: Dict[int, str]) -> None:
         raise AttributeError(
             "Setting the partition_id_to_natural_id dictionary is not allowed."
         )
+
+    def _check_supported_type_of_value_in_partition_by(self) -> None:
+        values = self.dataset[self._partition_by]
```

Comment:
I note this can take quite long. For example for our `"flwrlabs/shakespeare"` it takes ~9s. I'd propose only loading the first row only. wdyt?:
```
values = self.dataset.select(range(1))[self._partition_by]
```

Reply:
Yes, we can do that. In that case, the `np.object_` will be detected only if the `None` will be the first object, but that should work anyway.

Reply:
I removed that function now